### PR TITLE
refactor(api): application keys crypto

### PR DESCRIPTION
**Pull request by fsamin**

### CODE REVIEW

File: engine/api/application/dao_key.go [hunk 1]
```diff
@@ -2,129 +2,101 @@ package application
...
 	}
-	app.Keys = keys
-	return nil
+	isValid, err := gorpmapping.CheckSignature(k, k.Signature)
+	if err != nil {
+		return nil, err
+	}
+	if !isValid {
+		log.Error(context.Background(), "application.LoadKey> application key %d data corrupted", k.ID)
+		return nil, sdk.ErrNotFound
```
Comment (richardlt) [inline engine/api/application/dao_key.go]: wrap

File: engine/api/application/dao_key.go [hunk 2]
```diff
@@ -2,129 +2,101 @@ package application
...
-	for i := range res {
-		p := res[i]
-		keys[i] = sdk.ApplicationKey(p)
-		decrypted, err := secret.Decrypt([]byte(keys[i].Private))
-		if err != nil {
-			log.Error(context.TODO(), "LoadAllDecryptedKeys> Unable to decrypt private key %s/%s: %v", app.Name, keys[i].Name, err)
-		}
-		keys[i].Private = string(decrypted)
+	if !found {
+		return nil, sdk.ErrNotFound
```
Comment (richardlt) [inline engine/api/application/dao_key.go]: wrap

File: engine/api/application/gorp_model.go [hunk 3]
```diff
@@ -14,13 +14,24 @@ import (
...
-type dbApplicationKey sdk.ApplicationKey
+type dbApplicationKey struct {
+	gorpmapping.SignedEntity
+	sdk.ApplicationKey
+}
+
+func (e dbApplicationKey) Canonical() gorpmapping.CanonicalForms {
+	var _ = []interface{}{e.ApplicationID, e.ID, e.Name}
+	return gorpmapping.CanonicalForms{
+		"{{print .ApplicationID}} {{print .ID}} {{.Name}}",
```
Comment (richardlt) [inline engine/api/application/gorp_model.go]: Weed didn't add spaces for other entities

Comment (fsamin) [reply engine/api/application/gorp_model.go]: weed ?

Comment (bnjjj) [reply engine/api/application/gorp_model.go]: 🚬🔥

File: engine/api/action/loader.go [hunk 4]
```diff
@@ -3,6 +3,8 @@ package action
 import (
 	"context"
 
+	"github.com/ovh/cds/engine/api/group"
```
Comment (sguiheux) [inline engine/api/action/loader.go]: wrong place

File: engine/api/application/dao_key.go (hunk 1 above)
Comment (fsamin) [reply engine/api/application/dao_key.go]: done

File: engine/api/application/dao_key.go (hunk 2 above)
Comment (fsamin) [reply engine/api/application/dao_key.go]: done

File: engine/api/application/gorp_model.go (hunk 3 above)
Comment (fsamin) [reply engine/api/application/gorp_model.go]: done

File: engine/api/action/loader.go (hunk 4 above)
Comment (fsamin) [reply engine/api/action/loader.go]: done

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#12439.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✘  
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#12441.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
  * Docs ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Helm Tests -   
  * Standalone Tests ✘

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#12442.0 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
  * Docs ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Helm Tests -   
  * Standalone Tests ✘

**Comment by ovh-cds** [comment]
CDS Report build-all-cds#12442.1 ✘  
* Build
  * Build and Package All ✔ 
  * Build Debpacker ✔ 
  * Build OS-Ansible-Inventory ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test SDK Rust ✔ 
  * Test UI ✔ 
* Package
  * Docker ✔ 
  * Docs ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Helm Tests -   
  * Standalone Tests ✘

**Pull request merged**